UC-Berkeley-Extension/global
# Pardot contact

## Asana Ticket(s)
- See ticket [Pardot forms in Drupal](https://app.asana.com/0/1174970560648905/1196144381080481/f).

## Description
- Shows one model for displaying Pardot Form Handlers as Drupal blocks.

## Testing instructions

* Checkout this branch
* Inside the VM, run `drush cr` and `drush cim`
* Visit https://global.ddev.site/
* Observe the `Custom Pardot form` block
* Observe the hidden fields 
  - success_location, error_location, full_path, page_title
* Observe the value of `full_path` is set to `Look! A value!` by `/js/bglobal_pardot.js`

## Remaining Steps

* I am not convinced this is the best approach. Using an iFrame for Pardot forms directly still might be better.
* If we use a Form Handler -- https://help.salesforce.com/articleView?id=pardot_form_handlers.htm&type=5 -- then we need to ensure that our forms sync with Pardot fields. We can do that through the theme element introduced here.

In some respected, I repeated part of the FormAPI here. If we need to route the form through Drupal rather than making a direct request to Pardot, we would need to use the FormAPI instead.

- [ ] Determine how we should derive Program Name.
- [ ] Add actual Pardot tracking code.
